src/components/atoms/Modal/ModalFrame/index.tsx

```diff
@@ -21,7 +21,9 @@ const Modal: React.FC<Props> = ({ size, isVisible, onClose, children }) => {
     unmountOnExit: true,
   });
 
-  const handleClose = useCallback(() => onClose?.(), [onClose]);
+  const handleClose = useCallback(() => {
+    isVisible && onClose?.();
```

Comment:
```suggestion
    if (isVisible) onClose?.();
```

It is a matter of coding style preference, but I think the use of if statements would make the intent clearer here.